sound/soc/sof/pcm.c

```diff
@@ -363,6 +369,15 @@ static int sof_pcm_trigger(struct snd_pcm_substream *substream, int cmd)
 		stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_START;
 		break;
 	case SNDRV_PCM_TRIGGER_SUSPEND:
+		/*
+		 * Don't suspend the d0i3 compatible stream when target
+		 * is S0ix, and ignore the coming round of DAPM events.
+		 */
+		if (sstream->d0i3_compatible && !sdev->target_state) {
```

Comment:
better to be explicit here about target_state > ACPI_STATE_S0

Reply:
OK, that will be sdev->target_state == ACPI_STATE_S0.